liota/dccs/dcc.py

```diff
@@ -111,10 +111,11 @@ def publish(self, reg_metric):
             log.error("RegisteredMetric object is expected.")
             raise TypeError("RegisteredMetric object is expected.")
         message = self._format_data(reg_metric)
-        if hasattr(reg_metric, 'msg_attr'):
-            self.comms.send(message, reg_metric.msg_attr)
-        else:
-            self.comms.send(message, None)
+        if message:
```

Comment:
If there is no message, would it be better to raise an exception for the ABC implementers?

Reply:
This function is being called by send_thread() in liota_core, whenever a metric is supposed to be ready to send_data. It is not in the user thread, and hence, user will not be able to anything to catch this exception. Hence, it will not be good to throw this exception.